### PR TITLE
Add integration tests for transaction with relational scan

### PR DESCRIPTION
**Pull request by jnmt**
This PR adds more integration tests for the relational scan, especially from the transaction perspective. DistributedTransactionXX and TwoPhaseCommitTransactionXX cover the basic tests with relational scan. ConsensusCommitTransactionXX and TwoPhaseConsensusCommitTransactionXX include the overlap check to prevent scan-after-writes.

### CODE REVIEW

**Review by brfrn169 — APPROVED** [review]
LGTM! Thank you!

File: integration-test/src/main/java/com/scalar/db/transaction/consensuscommit/ConsensusCommitRelationalScanIntegrationTestBase.java [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+    assertThat(thrown).isInstanceOf(IllegalArgumentException.class);
+  }
+
+  @Test
+  public void scan_PutResultNonOverlappedWithRelationalScanGiven_ShouldThrowException()
+      throws TransactionException {
+    // Arrange
+    populateRecords();
+    DistributedTransaction transaction = manager.start();
+    Put put = preparePut(10, NUM_TYPES);
```
Comment (komamitsu) [inline integration-test/src/main/java/com/scalar/db/transaction/consensuscommit/ConsensusCommitRelationalScanIntegrationTestBase.java:116]: Let me confirm this a little bit. If `preparePut` uses the account type as a clustering key as well like `{key: (10, 4), value: 4}`,   the new record wouldn't match any case of https://github.com/scalar-labs/scalardb/blob/eca907801a6561a1eb8dac9dcd92a8ef383463b4/core/src/main/java/com/scalar/db/transaction/consensuscommit/Snapshot.java#L279-L281 ?

**Review by feeblefakie — APPROVED** [review]
LGTM! Thank you!

File: integration-test/src/main/java/com/scalar/db/transaction/consensuscommit/ConsensusCommitRelationalScanIntegrationTestBase.java (hunk 1 above)
Comment (jnmt) [reply integration-test/src/main/java/com/scalar/db/transaction/consensuscommit/ConsensusCommitRelationalScanIntegrationTestBase.java:116]: Thank you for the question. Actually, the table schema in `ConsensusCommitRelationalScanIntegrationTestBase` is a little different from `ConsensusCommitIntegrationTestBase`. All account IDs are unique, and there are no clustering keys, i.e., the account type is just a regular column. So, the above test covers case 2, the previous covers case 1, and the next covers case 3.

File: integration-test/src/main/java/com/scalar/db/api/DistributedTransactionRelationalScanIntegrationTestBase.java [hunk 2]
```diff
@@ -0,0 +1,369 @@
...
+      // Act Assert
+      Assertions.assertThatCode(
+              () -> {
+                DistributedTransaction tx = managerWithDefaultNamespace.start();
+                tx.scan(scan);
+                tx.commit();
+              })
+          .doesNotThrowAnyException();
+    } finally {
+      if (managerWithDefaultNamespace != null) {
```
[the remark this answers is not in the document]
Comment (jnmt) [reply integration-test/src/main/java/com/scalar/db/api/DistributedTransactionRelationalScanIntegrationTestBase.java]: Thank you! Fixed in bf03584.

File: integration-test/src/main/java/com/scalar/db/transaction/consensuscommit/ConsensusCommitRelationalScanIntegrationTestBase.java (hunk 1 above)
Comment (komamitsu) [reply integration-test/src/main/java/com/scalar/db/transaction/consensuscommit/ConsensusCommitRelationalScanIntegrationTestBase.java:116]: Got it! Thank you!

**Review by komamitsu — APPROVED** [review]
LGTM! 👍

**Comment by jnmt** [comment]
@Torch3333 I will merge this now and probably continue to add you as a reviewer just for your reference even while you are away. Sorry, it might be a bit noisy, though.

**Review by Torch3333 — COMMENTED** [review]
LGTM, thank you!

**Pull request merged**